### PR TITLE
Recover shortcuts on local path

### PR DESCRIPTION
**Pull request by dnesbitt61**
Working version (at least on a couple test routes). Still need to update pathalgorithm.cc to use it and figure out performance impact.

### CODE REVIEW

**Comment by kevinkreiser** [comment]
looks good to me. is there any reason why this is in a seprate source file?

File: src/thor/formlocalpath.cc [hunk 1]
```diff
@@ -118,5 +136,99 @@ if (!(directededge->endnode() == prior_local_node)) {
...
+  const NodeInfo* nodeinfo = tile->node(endnode);
+  const PointLL& endll = nodeinfo->latlng();
+
+  // Expand from the start node
+  tile = graphreader.GetGraphTile(startnode);
+  nodeinfo = tile->node(startnode);
+  GraphId edgeid(startnode.tileid(), startnode.level(),
+                 nodeinfo->edge_index());
+  const DirectedEdge* directededge = tile->directededge(nodeinfo->edge_index());
+  for (uint32_t i = 0, n = nodeinfo->edge_count(); i < n;
```
Comment (kevinkreiser) [inline src/thor/formlocalpath.cc]: just a style thing but why instantiate n? just use nodeinfo->edge_count() directly, i'm certain its inlined. this way reads poorly (more like c?) in my opinion

**Comment by dgearhart** [comment]
:+1:

File: src/thor/formlocalpath.cc (hunk 1 above)
Comment (dnesbitt61) [inline src/thor/formlocalpath.cc]: just a habit I got into...

**Comment by dnesbitt61** [comment]
I put this in a separate file since I was not sure how large and complex it might be and whether we would actually keep it.

File: src/thor/formlocalpath.cc (hunk 1 above)
Comment (kevinkreiser) [inline src/thor/formlocalpath.cc]: no big deal just thought id mention it in case you were doing it subconciously

**Comment by kevinkreiser** [comment]
cool makes sense

**Pull request merged**